HW06-Annotation/src/main/java/com/testframework/TestFramework.java

```diff
@@ -0,0 +1,98 @@
+package com.testframework;
+
+import com.akproductions.After;
+import com.akproductions.Before;
+import com.akproductions.Test;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class TestFramework {
+    Class clazz;
+    public TestFramework (String classToTest) throws Exception {
+        clazz = Class.forName(classToTest);
+    }
+
+    public String doTest () throws Exception
+    {
+        Method[] methodsAll = clazz.getDeclaredMethods();
+        var methodsBefore = new ArrayList<Method>(10);
+        var methodsAfter = new ArrayList<Method>(10);
+        var methodsTest = new ArrayList<Method>(10);
+        AtomicInteger testSuccess= new AtomicInteger();
+        AtomicInteger testError= new AtomicInteger();
+        for (Method method : methodsAll) {
+            Annotation[] annotations = method.getDeclaredAnnotations();
+            for (Annotation annotation : annotations) {
+                if (annotation.annotationType().equals(Before.class)) methodsBefore.add(method);
+                if (annotation.annotationType().equals(After.class)) methodsAfter.add(method);
+                if (annotation.annotationType().equals(Test.class)) methodsTest.add(method);
+            }
+            }
+        System.out.println("[INFO] --- Before methods:");
+        methodsBefore.forEach(c ->   System.out.println(c.getName()));
+        System.out.println("[INFO] --- After methods:");
+        methodsAfter.forEach(c ->   System.out.println(c.getName()));
+        System.out.println("[INFO] --- Test methods:");
+        methodsTest.forEach(c ->   System.out.println(c.getName()));
+
+        //make tests
+        methodsTest.forEach(methodTest ->
+        {
+            System.out.println("[INFO] --- run test for  " + methodTest.getName());
+            try {
+                Constructor constructor = clazz.getConstructor();
+                var object = constructor.newInstance();
+                    methodsBefore.forEach(methodBefore ->
+                    {
+                        System.out.println("[DEBUG] --- invoke before method  " + methodBefore.getName());
+                        methodBefore.setAccessible(true);
+                        try {
+                            methodBefore.invoke(object);
+                        } catch (Exception e) {
+                            System.out.println("[ERROR] --- error in calling before method  " + methodBefore.getName());
+                            e.printStackTrace(System.out);
+                        }
+                    });
```

Comment:
Можно выделить в метод:
```suggestion
                    runMethods(methodsBefore, object);
```

Для After-методов то же самое.